### ISSUE TITLE
Feature Request: Curve Plugin Support

### ISSUE THREAD

**Issue by TheKingOfAtlantis** [Rem0o/FanControl.Releases#751]
While I love the currently available support for plugins (especially since I'm using a commander pro) extending that API to cover curves would be fantastic, as personally, I'd love to create custom curves which fill a couple of holes that I feel are missing

p.s. are there any plans to make the project open source?

**Comment by Rem0o** [comment]
I thought about it, but decided not to, as it would be quite complex.

Curves have many aspects:

1) UI => I would need to expose the whole UI library/code of FanControl for a nice integration
2) Internal logic => Curves have access to a whole bunch of internal dependencies, like the dialogservice, logger, temperature sources, and so on.  All of these would need to be exposed, which is basically the whole backend.
3) Serialization (!!!) => That's also a big one. Each curve needs to be serialized and deserialized properly.

TLDR: allowing curve injection is basically exposing the whole software backend in the plugin system, which at that point, as you guessed it, OSS.

If you want to see the discussion on that subject.
https://github.com/Rem0o/FanControl.Releases/issues/89

**Comment by Rem0o** [comment]
If you do have a specific curve suggestion, feel free to share it. Otherwise, will close this for now.
